File: src/main/java/com/droiddevgeeks/crashsink/DeviceMetadata.java
package com.droiddevgeeks.crashsink;

/** Immutable device/app metadata, collected once off the crash path. Pure Java (no Android). */
public final class DeviceMetadata {
    public final String osVersion;
    public final int sdkInt;
    public final String manufacturer;
    public final String model;
    public final String appVersionName;
    public final int appVersionCode;

    public DeviceMetadata(final String osVersion, final int sdkInt, final String manufacturer,
                          final String model, final String appVersionName, final int appVersionCode) {
        this.osVersion = osVersion;
        this.sdkInt = sdkInt;
        this.manufacturer = manufacturer;
        this.model = model;
        this.appVersionName = appVersionName;
        this.appVersionCode = appVersionCode;
    }
}

File: crashsink/src/main/kotlin/com/droiddevgeeks/crashsink/DeviceMetadata.kt
package com.droiddevgeeks.crashsink

/**
 * Immutable device/app metadata, collected once off the crash path. Pure (no Android).
 *
 * Internal: consumers never see this type. It is serialized to the `contexts` JSON string the
 * [CrashSink] receives; it is not part of the consumer-facing API.
 */
internal class DeviceMetadata(
    // osVersion/manufacturer/model come from android.os.Build and may legitimately be null
    // (and are null under host unit tests), so they stay nullable.
    val osVersion: String?,
    val sdkInt: Int,
    val manufacturer: String?,
    val model: String?,
    val appVersionName: String,
    val appVersionCode: Int
)
